apps/src/codebridge/FileBrowser/index.tsx

```diff
@@ -48,11 +55,59 @@ const InnerFileBrowser = React.memo(
     moveFilePrompt,
     renameFilePrompt,
     renameFolderPrompt,
-    toggleFileVisibility,
+    setFileVisibility,
+    setFileIsValidation,
   }: FilesComponentProps) => {
     const {openFile, deleteFile, toggleOpenFolder, deleteFolder} =
       useCodebridgeContext();
     const isStartMode = getAppOptionsEditBlocks() === START_SOURCES;
+    const hasValidationFile = Object.values(files).some(f => f.validation);
+
+    const handleSetFileVisibility = (fileId: string, hidden: boolean) => {
+      setFileVisibility(fileId, hidden);
+      // When we set the visibility, we also want to set the file to validation: false.
+      setFileIsValidation(fileId, false);
+    };
+
+    const startModeFileDropdownOptions = (file: ProjectFile) => {
+      // We only support one validation file per project, so if we already have one,
+      // do not show the option to mark another file as validation.
+      const options = [];
+      if (!file.validation && !hasValidationFile) {
```

Comment:
Is it possible for the current file to be validation if there is no validation file? In other words, why is the `!file.validation` check needed?

Reply:
nope you're right, it's a redundant check